tsl/src/fdw/data_node_scan_plan.c

```diff
@@ -955,6 +1652,54 @@ data_node_scan_path_create(PlannerInfo *root, RelOptInfo *rel, PathTarget *targe
 	return &scanpath->cpath.path;
 }
 
+/*
+ * data_node_join_path_create
+ *	  Creates a path corresponding to a scan of a foreign join,
+ *	  returning the pathnode.
+ *
+ * There is a usually-sane default for the pathtarget (rel->reltarget),
+ * so we let a NULL for "target" select that.
+ *
+ * The code is based on PostgreSQL's create_foreign_join_path function.
+ */
+static Path *
+data_node_join_path_create(PlannerInfo *root, RelOptInfo *rel, PathTarget *target, double rows,
+						   Cost startup_cost, Cost total_cost, List *pathkeys,
+						   Relids required_outer, Path *fdw_outerpath, List *private)
+{
+	DataNodeScanPath *scanpath = palloc0(sizeof(DataNodeScanPath));
+
+	if (rel->lateral_relids && !bms_is_subset(rel->lateral_relids, required_outer))
+		required_outer = bms_union(required_outer, rel->lateral_relids);
+
+	/*
+	 * We should use get_joinrel_parampathinfo to handle parameterized paths,
+	 * but the API of this function doesn't support it, and existing
+	 * extensions aren't yet trying to build such paths anyway.  For the
+	 * moment just throw an error if someone tries it; eventually we should
+	 * revisit this.
+	 */
+	if (!bms_is_empty(required_outer) || !bms_is_empty(rel->lateral_relids))
+		elog(ERROR, "parameterized foreign joins are not supported yet");
```

Comment:
Add test case?

Reply:
These parameters are replaced before the `data_node_scan_plan` is generated in our implementation; this check is from PG upstream. However, I added a test case for a prepared statement to test that the join pushdown also works in this case. 

We have a similar check also for [regular scans](https://github.com/timescale/timescaledb/blob/206056ca12c611b2038b35bfa0eac43372a563c2/tsl/src/fdw/data_node_scan_plan.c#L937) in place. I wanted to check if we were covering this line in the existing tests. But unfortunately, codecov currently has [an outage](https://status.codecov.com/) at the moment and I can't access the reports.

Reply:
@erimatnor Codecov works now and I checked the coverage of the closely related line in the existing code. [This line](https://github.com/timescale/timescaledb/blob/206056ca12c611b2038b35bfa0eac43372a563c2/tsl/src/fdw/data_node_scan_plan.c#L937) in the existing `data_node_scan_path_create` function is also one of these lines that are copied from PG upstream and cannot directly be tested in our implementation. So, it is hard to create a test case for the new/similar line. 

https://app.codecov.io/gh/timescale/timescaledb/commit/0562c6d89aa757e7aa7f3acfaea3f3d386fe9f6c/blob/tsl/src/fdw/data_node_scan_plan.c#L1635